### ISSUE TITLE
Problems with bento/centos-8

### ISSUE THREAD

**Issue by 5laps2go** [josenk/vagrant-vmware-esxi#122]
When I use this plugin with some vagrant boxes like bento/centos-8, the following error is appeared:
```
==> bento-centos-8: Successfully added box 'bento/centos-8' (v202110.25.0) for 'vmware_desktop'!
==> bento-centos-8: Virtual Machine will be built.
VMware ovftool 4.4.1 (build-16812187)
==> bento-centos-8: ---   --- ESXi Summary ---
==> bento-centos-8: --- ESXi host       : 192.168.1.31
==> bento-centos-8: --- Virtual Network : ["VM Network"]
==> bento-centos-8: --- Disk Store      : datastore1
==> bento-centos-8: --- Resource Pool   : /
==> bento-centos-8: ---  --- Guest Summary ---
==> bento-centos-8: --- VM Name         : bento-centos-8
==> bento-centos-8: --- Box             : bento/centos-8
==> bento-centos-8: --- Box Ver         : 202110.25.0
==> bento-centos-8: --- Memsize (MB)    : 1024
==> bento-centos-8: --- CPUS            : 1
==> bento-centos-8: --- Guest OS type   : centos-64
==> bento-centos-8: --- Allow Overwrite : True
==> bento-centos-8: ---   --- Guest Build ---
Opening VMX source: /home/vagrant/.vagrant.d/boxes/bento-VAGRANTSLASH-centos-8/202110.25.0/vmware_desktop/ZZZZ_bento-centos-8.vmx
Error: File (/Users/tsmith/.cache/packer/aec6714f7b2ba1e242cbc4196e9d1366dd92d460.iso) could not be found.
Completed with errors
There was an OVF Tool error.
```
The iso file is defined as cdrom-image in the VMX file as follows and causes ovftool error.  
```
ide0:0.devicetype = "cdrom-image"
ide0:0.filename = "/Users/tsmith/.cache/packer/aec6714f7b2ba1e242cbc4196e9d1366dd92d460.iso"
ide0:0.present = "TRUE"
```

I have checked to omit the error by modifying devicetype = "cdrom-image" to devicetype = "cdrom-raw".
Can you replace the "cdrom-image"?

**Comment by jmcgnh** [comment]
You're not user "tsmith", right, and you probably don't have a file with
that path name on your windows filesystem. My guess is that Tim Smith from
the Bento project inadvertently left the CD-Device configured when packing
the box. The easy thing to do would be to move on to a different box,
possibly from a different box provider - these base boxes are meant to be
pretty generic - to try to find one that doesn't have this particular bug
in its build.

You could also report your problem by filing an issue at the Bento Github:
https://github.com/chef/bento/issues

 -- jmcg

**Comment by 5laps2go** [comment]
I know this is bento package issue.   Actually, I used to use bento box for several months. I found the issue when  I update the box today.    I also tried several vagrant boxes and found some vagrant boxes have the same issue.
 
I can fix the box issue easily by adding a few lines of codes in vagrant-vmware-esxi-2.5.3/lib/vagrant-vmware-esxi/action/createvm.rb
```
            when /^ethernet[0-9]/i
              # Just save the nic type
              if (line =~ /ethernet0.virtualDev = /i) and desired_nic_type.nil?
                desired_nic_type = line.gsub(/ethernet0.virtualDev = /i, '').gsub('"', '').strip
              end

+            when /\.devicetype = "cdrom-image"/i
+              new_vmx_contents << line.gsub(/cdrom-image/, 'cdrom-raw')
+
            else
              new_vmx_contents << line
            end
```

Anyway, I'll report the problem to each box team.   Please close this issue.
Thank you.

**Comment by josenk** [comment]
Thanks for the suggestion.   I think it's a good suggestion for a fix.   It seems this issue is popping up more frequently.

**Comment by josenk** [comment]
Thanks for the suggestion.   I think it's a good suggestion for a fix.   It seems the bug/issue is popping up more and more requently.

**Comment by josenk** [comment]
Fixed in v2.5.6
